react-backend/client/src/components/hoc/admin_login_check.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const requestStates = {
+  SUCCESS: 'SUCCESS',
+  PENDING: 'PENDING',
+};
+const adminLoginCheck = (Child) => {
+  class AdminLoginCheck extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        auth: requestStates.PENDING,
+        errorMessage: '',
+      };
+    }
+
+    componentDidMount() {
+      const adminToken = this.props.auth;
+      if (!adminToken) throw new Error('No admin token');
+
+      const headers = {
+        Authorization: adminToken,
+        'Content-Type': 'application/json',
+      };
+      const opts = { headers, method: 'post' };
+
+      new Promise((resolve, reject) => {
+        this.cancelRequest = () => {
+          reject(new Error('REQUEST_CANCELLED'));
+        };
+        fetch('/api/admin/check', opts)
+          .then(resolve)
+          .catch(reject);
+      })
+        .then(() => this.setState({ auth: requestStates.SUCCESS }))
+        .catch((error) => {
+          if (error.message === 'REQUEST_CANCELLED') return;
```

Comment:
https://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects

Reply:
OK, well I googled "check if component is still mounted before doing something react" and [this is the first result](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html)

Reply:
Thanks, this seems to have fairly similar functionality, but with mine cancelling immediately rather than waiting for the fetch to return. Although the article still doesn't answer the question of where to store the cancellable promise.

TBH I don't see the need to pick between either as there's the eagerness to move to redux in the long run.